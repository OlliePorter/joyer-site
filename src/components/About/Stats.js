import React from 'react';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import HomeIcon from '@material-ui/icons/Home';
import SentimentDissatisfiedIcon from '@material-ui/icons/SentimentDissatisfied';
import PhoneDisabledIcon from '@material-ui/icons/PhoneDisabled';import { makeStyles } from '@material-ui/core/styles';
import theme from '../../theme';

const useStyles = makeStyles({
  coverBackground: {
    backgroundColor: theme.palette.primary.light,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: '40vh',
    width: '100%',
    overflow: 'hidden',
    [theme.breakpoints.down('md')]: {
      minHeight: '40.1em',
    },
  },
  coverPadding: {
    [theme.breakpoints.up('md')]: {
      marginLeft: '200px',
      marginRight: '200px',
    },
  },
  statsColumnContainer: {
    display: 'flex',
    flexDirection: 'column',
    maxWidth: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    padding: '3em',
    [theme.breakpoints.down('md')]: {
      padding: '2em',
    },
  },
  statsText: {
    color: 'black',
    fontWeight: 300,
    [theme.breakpoints.down('md')]: {
      paddingLeft: '30px',
      paddingRight: '30px',
      textAlign: 'center',
    },
  },
  statsTextTitle: {
    color: 'black',
    textAlign: 'center',
    fontWeight: 300,
    paddingTop: '2em',
  },
  statsTextSubtitle: {
    color: 'black',
    textAlign: 'center',
    fontWeight: 300,
    paddingTop: '0.5em',
  },
  testimonialImage: {
    width: '60px',
    height: '60px',
    borderRadius: '50%',
    justifyContent: 'center',
  },
  statsIcon: {
    color: theme.palette.primary.main,
    transform: 'scale(2.3)',
  },
  statsGrid: {
    [theme.breakpoints.down('md')]: {
      display: 'flex',
      flexDirection: 'row',
    },
  }
});

export default function Stats() {
    const classes = useStyles();

    return (
      <>
        <div className={classes.coverBackground}>
            <div className={classes.coverPadding}>
              <Grid container spacing={6} className={classes.statsGrid}>
                <Grid item xs={4}>
                  <div className={classes.statsColumnContainer}>
                    <HomeIcon className={classes.statsIcon}/>
                    <Typography className={classes.statsTextTitle} component="h2" variant="h5">
                      000,000s
                    </Typography>
                    <Typography className={classes.statsTextSubtitle} variant="body1">
                      of people stuck inside.
                    </Typography>
                  </div>
                </Grid>
                <Grid item xs={4}>
                  <div className={classes.statsColumnContainer}>
                    <SentimentDissatisfiedIcon className={classes.statsIcon}/>
                    <Typography className={classes.statsTextTitle} component="h2" variant="h5">
                      1 in 4
                    </Typography>
                    <Typography className={classes.statsTextSubtitle} variant="body1">
                      experience mental health problems.
                    </Typography>
                  </div>
                </Grid>
                <Grid item xs={4}>
                  <div className={classes.statsColumnContainer}>
                    <PhoneDisabledIcon className={classes.statsIcon}/>
                    <Typography className={classes.statsTextTitle} component="h2" variant="h5">
                      60%
                    </Typography>
                    <Typography className={classes.statsTextSubtitle} variant="body1">
                      of people end conversations in less than 90 seconds.
                    </Typography>
                  </div>
                </Grid>
              </Grid>
            </div>
        </div>
      </>
    );
}
