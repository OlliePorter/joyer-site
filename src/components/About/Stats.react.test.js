import React from 'react';
import Stats from './Stats';
import renderer from 'react-test-renderer';

it('Cover renders correctly', () => {
    const tree = renderer.create(
      <Stats />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });