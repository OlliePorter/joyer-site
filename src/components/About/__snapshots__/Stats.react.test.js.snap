// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cover renders correctly 1`] = `
<div
  className="makeStyles-coverBackground-1"
>
  <div
    className="makeStyles-coverPadding-2"
  >
    <div
      className="MuiGrid-root makeStyles-statsGrid-9 MuiGrid-container MuiGrid-spacing-xs-6"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          className="makeStyles-statsColumnContainer-3"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-statsIcon-8"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
            <path
              d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
          <h2
            className="MuiTypography-root makeStyles-statsTextTitle-5 MuiTypography-h5"
          >
            9 hours
          </h2>
          <p
            className="MuiTypography-root makeStyles-statsTextSubtitle-6 MuiTypography-body1"
          >
            The time people spend consuming media on social platforms each day.
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          className="makeStyles-statsColumnContainer-3"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-statsIcon-8"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="15.5"
              cy="9.5"
              r="1.5"
            />
            <circle
              cx="8.5"
              cy="9.5"
              r="1.5"
            />
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-3.5c.73 0 1.39.19 1.97.53.12-.14.86-.98 1.01-1.14-.85-.56-1.87-.89-2.98-.89-1.11 0-2.13.33-2.99.88.97 1.09.01.02 1.01 1.14.59-.33 1.25-.52 1.98-.52z"
            />
          </svg>
          <h2
            className="MuiTypography-root makeStyles-statsTextTitle-5 MuiTypography-h5"
          >
            1 in 4
          </h2>
          <p
            className="MuiTypography-root makeStyles-statsTextSubtitle-6 MuiTypography-body1"
          >
            People experience mental health problems each year.
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          className="makeStyles-statsColumnContainer-3"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-statsIcon-8"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2c-5.52 0-10 4.48-10 10s4.48 10 10 10 10-4.48 10-10-4.48-10-10-10zm3.61 6.34c1.07 0 1.93.86 1.93 1.93 0 1.07-.86 1.93-1.93 1.93-1.07 0-1.93-.86-1.93-1.93-.01-1.07.86-1.93 1.93-1.93zm-6-1.58c1.3 0 2.36 1.06 2.36 2.36 0 1.3-1.06 2.36-2.36 2.36s-2.36-1.06-2.36-2.36c0-1.31 1.05-2.36 2.36-2.36zm0 9.13v3.75c-2.4-.75-4.3-2.6-5.14-4.96 1.05-1.12 3.67-1.69 5.14-1.69.53 0 1.2.08 1.9.22-1.64.87-1.9 2.02-1.9 2.68zM11.99 20c-.27 0-.53-.01-.79-.04v-4.07c0-1.42 2.94-2.13 4.4-2.13 1.07 0 2.92.39 3.84 1.15-1.17 2.97-4.06 5.09-7.45 5.09z"
            />
          </svg>
          <h2
            className="MuiTypography-root makeStyles-statsTextTitle-5 MuiTypography-h5"
          >
            Quantity &gt; Quality
          </h2>
          <p
            className="MuiTypography-root makeStyles-statsTextSubtitle-6 MuiTypography-body1"
          >
            Apps today promote extensive friend lists over of high quality relationships.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
